app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -240,7 +263,8 @@ def host_guest_devices(persister_hardware, host, nics, networks)
 
       unless network.nil?
         switch_uid = network.try(:id) || network.name
-        attributes[:switch] = persister.host_virtual_switches.lazy_find(:host => persister_host, :uid_ems => switch_uid)
+        distributed_virtual_switch = persister.distributed_virtual_switches.lazy_find(:host => persister_host, :uid_ems => switch_uid)
+        attributes[:switch] = persister.host_switches.lazy_find(:host => persister_host, :switch => distributed_virtual_switch)
```

Comment:
I'd have to check but I think guest_device.switch is not a host_switch but just the `distributed_virtual_switch`